extensions/s3/src/test/java/com/hazelcast/jet/s3/S3TestBase.java

```diff
@@ -168,6 +170,23 @@ void deleteBucket(S3Client client, String bucket) {
         }
     }
 
+    Stream<String> S3ObjectToLines(S3Object o, S3Client client, String bucketName) {
+        try {
+            ResponseInputStream<GetObjectResponse> is = client.getObject(req -> req.bucket(bucketName).key(o.key()), toInputStream());
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, CHARSET))) {
+                // materialize the stream, since we can't read it afterwards
+                return reader.lines().collect(Collectors.toList()).stream();
+            } catch (IOException e) {
+                throw new AssertionError("Error reading file ", e);
+            }
+        } catch (NoSuchBucketException e) {
```

Comment:
the exception from the issue is `NoSuchKeyException` not `NoSuchBucketException`
I think we can use the parent `S3Exception` here